src/Thermal/RadiativeFB.cpp

```diff
@@ -0,0 +1,308 @@
+  r_planet *= rsun;
+  r_bdwarf *= rsun;
+  r_star   *= rsun;
+}
+
+//=================================================================================================
+//  SinkHeating::SinkHeating()
+/// SinkHeating class destructor.
+//=================================================================================================
+template <int ndim>
+SinkHeating<ndim>::~SinkHeating()
+{
+
+}
+
+//=================================================================================================
+//  SinkHeating::SinkTemperature()
+/// Returns the sink temperature which is found via it's luminosity, such that:
+/// T = (L / (4 * pi * boltz * R^2))^1/4
+//=================================================================================================
+template <int ndim>
+FLOAT SinkHeating<ndim>::SinkTemperature
+(FLOAT L,
+ FLOAT r)
+{
+  return pow(L / (4.0 * pi * rad_const * r * r), 0.25);
+}
+
+//=================================================================================================
+//  SinkHeating::SinkLuminosity()
+/// Returns the sink luminosity which consists of two parts: an intrinsic luminosity which only
+/// applies to stars (objects M > 80 Mjup) and an accretion luminosity. Specifically
+/// L = f_n * (m / m_sun)^3 * L_sun + f_acc * (G * m * m_dot / r) * (1 - (r / 2 * rsink))
+/// f_n is the flag for intrinsic accretion. f_acc is the fraction of gravitational energy
+/// converted to heat energy (typically f_acc = 0.75, Offner et al. (2010)). R is the radius of
+/// the object at the center of the sink and Rsink is the radius of the sink itself.
+//=================================================================================================
+template <int ndim>
+FLOAT SinkHeating<ndim>::SinkLuminosity
+(FLOAT m,
+ FLOAT mdot,
+ FLOAT rsink,
+ FLOAT r,
+ int f_n)
+{
+  return f_n * pow(m / msun, 3.0) * lsun +
+         f_acc * ((grav_const * m * mdot) / r) * (1 - (r / (2.0 * rsink)));
+}
+
+//=================================================================================================
+//  SinkHeating::AmbientTemp()
+/// Calculates the ambient temperature of particles assuming continuous radiative feedback, i.e.
+/// when particles are accreted onto an object, the energy is instantly released into the system.
+//=================================================================================================
+template <int ndim>
+FLOAT SinkHeating<ndim>::AmbientTemp
+(Particle<ndim> &part,
+ Sinks<ndim> *sinks)
+{
+  FLOAT temp = 0.0;
+
+  for (int i = Ncentral; i < sinks->Nsink; ++i) {
+    SinkParticle<ndim> sink = sinks->sink[i];
+
+    FLOAT dist      = Distance(part.r, sink.star->r, ndim);
+    FLOAT sink_m    = sink.star->m;
+    FLOAT sink_dmdt = sink.dmdt;
+    FLOAT sink_r    = sink.radius;
+
+    // Set source radius and intrinsic luminosity flag depending on sink mass
+    FLOAT r_source = r_planet;
+    int f_n = 0;
+    if (sink_m >= 13.0 * mjup) r_source = r_bdwarf;
```

Comment:
What are these 'random' constants?  Perhaps best to put them in their own well-named variable or simply add a comment explaining the constants.  I can guess what they are but best be explicit!

Reply:
I've added some comments to the code for now.